service-registry/src/main/java/io/servicecomb/serviceregistry/consumer/MicroserviceVersions.java

```diff
@@ -109,25 +109,20 @@ public void pullInstances() {
     setInstances(pulledInstances);
   }
 
-  private void setInstances(List<MicroserviceInstance> pulledInstances) {
-    synchronized (lock) {
-      instances = pulledInstances
-          .stream()
-          .filter(instance -> {
-            return MicroserviceInstanceStatus.UP.equals(instance.getStatus());
```

Comment:
It's better a add this logic here and filter out all other env's instances. Because this is a required feature, and filters are optional ones.